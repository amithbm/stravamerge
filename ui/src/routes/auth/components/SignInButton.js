import React from 'react'
import { Button } from 'react-bootstrap'

export default () => {
    const clientId = "37978"
    const redirectUri = process.env.NODE_ENV === 'production' ? 'https://rideanalysis.azurewebsites.net/tokenresponse' : "http://localhost:3000/tokenresponse"
    const authUri = "https://www.strava.com/oauth/authorize?client_id=" + clientId + "&response_type=code&redirect_uri=" + redirectUri + "&approval_prompt=auto";

    return  <Button bsSize="large" block href={authUri}>Sign In With Strava</Button>
}
