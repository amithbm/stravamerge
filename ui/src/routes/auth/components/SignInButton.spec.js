import React from 'react'
import {shallow} from 'enzyme'
import SignInButton from './SignInButton'
import toJson from 'enzyme-to-json'

describe('components', () => {
    describe('<SignInButton />', () => {
        it('renders as expected', () => {
            const component =  shallow(<SignInButton />)
            const tree = toJson(component)
            expect(tree).toMatchSnapshot()
        })
    })
})