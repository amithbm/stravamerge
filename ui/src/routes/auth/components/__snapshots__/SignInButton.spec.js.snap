exports[`components <SignInButton /> renders as expected 1`] = `
<Button
  active={false}
  block={true}
  bsClass="btn"
  bsSize="large"
  bsStyle="default"
  disabled={false}
  href="https://www.strava.com/oauth/authorize?client_id=37978&response_type=code&redirect_uri=http://localhost:3000/tokenresponse&approval_prompt=auto">
  Sign In With Strava
</Button>
`;
